### PR TITLE
created HashUtil

### PR DESCRIPTION
**Pull request by ilya449**

### CODE REVIEW

**Review by pavlokhomiak — APPROVED** [review]

File: src/main/resources/init_db.sql [hunk 1]
```diff
@@ -118,5 +118,8 @@ CREATE TABLE `internet_shop`.`shopping_carts_products` (
 ENGINE = InnoDB
 DEFAULT CHARACTER SET = utf8;
 
+ALTER TABLE `internet_shop`.`users`
+    ADD COLUMN `salt` VARBINARY(16) NOT NULL AFTER `password`;
```
Comment (KseniiaMakarova) [inline src/main/resources/init_db.sql]: or we can modify CREATE statement

File: src/main/java/com/internet/shop/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        StringBuilder hashedPassword = new StringBuilder();
+        try {
+            MessageDigest messageDigest = MessageDigest.getInstance(ALGORITHM);
+            messageDigest.update(salt);
+            byte[] digests = messageDigest.digest(password.getBytes());
+            for (byte b : digests) {
+                hashedPassword.append(String.format("%02x", b));
+            }
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Can't find algorithm " + ALGORITHM + e);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/util/HashUtil.java]:
```suggestion
            throw new RuntimeException("Can't find algorithm " + ALGORITHM, e);
```

File: src/main/java/com/internet/shop/util/HashUtil.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        }
+        return hashedPassword.toString();
+    }
+
+    public static boolean isPassValid(String password, User user) {
+        return hashPassword(password, user.getSalt())
+                .equals(user.getPassword());
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/util/HashUtil.java]: let's make this method private in AuthenticationServiceImpl

File: src/main/java/com/internet/shop/controller/InjectDataController.java [hunk 4]
```diff
@@ -1,46 +0,0 @@
...
-    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
-            throws ServletException, IOException {
-        User bob = new User("Bob", "bob123", "passBob", Set.of(Role.of("USER")));
-        User admin = new User("admin", "admin123", "admin123", Set.of(Role.of("ADMIN")));
-        User alice = new User("Alice", "alice123", "passAlice", Set.of(Role.of("USER")));
-        User userDave = new User("Dave", "dave3459", "passDave", Set.of(Role.of("USER")));
-        User userCharlie = new User("Charlie", "charlie777", "passCharlie",
-                Set.of(Role.of("USER")));
-        userService.create(bob);
-        userService.create(admin);
```
Comment (KseniiaMakarova) [inline src/main/java/com/internet/shop/controller/InjectDataController.java]: how can you create admin at the moment?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/resources/init_db.sql (hunk 1 above)
Comment (ilya449) [reply src/main/resources/init_db.sql]: Agree. And it will be more logically

File: src/main/java/com/internet/shop/util/HashUtil.java (hunk 3 above)
Comment (ilya449) [reply src/main/java/com/internet/shop/util/HashUtil.java]: Done

File: src/main/java/com/internet/shop/controller/InjectDataController.java (hunk 4 above)
Comment (ilya449) [reply src/main/java/com/internet/shop/controller/InjectDataController.java]: My mistake, I shouldn't delete this

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**